tests/pykafka/test_balancedconsumer.py

```diff
@@ -195,6 +201,36 @@ def test_external_kazoo_client(self):
         messages = [msg for msg in consumer]
         consumer.stop()
 
+    def test_zk_conn_lost(self):
+        """Check we restore zookeeper nodes correctly after connection loss
+
+        See also github issue #204.
+        """
+        zk = KazooClient(self.kafka.zookeeper)
+        zk.start()
+        try:
+            topic = self.client.topics[self.topic_name]
+            consumer = topic.get_balanced_consumer(b'test_zk_conn_lost',
+                                                   zookeeper=zk)
+            self.assertTrue(consumer._check_held_partitions())
+            zk.stop()  # expires session, dropping all our nodes
+
+            # Start a second consumer on a different zk connection
+            other_consumer = topic.get_balanced_consumer(b'test_zk_conn_lost')
+
+            zk.start()
```

Comment:
Does this actually trigger one of the watches that `consumer` has registered? If so, I totally understand what this test is doing, but I can't see exactly which watch it would trigger.